tools/src/main/java/org/apache/kafka/trogdor/coordinator/Coordinator.java

```diff
@@ -0,0 +1,337 @@
+            NodeManager nodeManager = nodeManagers.get(affectedNode);
+            if (nodeManager == null) {
+                nonexistentNodes.add(affectedNode);
+            } else {
+                affectedManagers.add(nodeManager);
+                nodeNames.add(affectedNode);
+            }
+        }
+        if (!nonexistentNodes.isEmpty()) {
+            log.warn("Fault {} refers to {} non-existent node(s): {}", fault.id(),
+                    nonexistentNodes.size(), Utils.join(nonexistentNodes, ", "));
+        }
+        log.info("Applying fault {} on {} node(s): {}", fault.id(),
+                nodeNames.size(), Utils.join(nodeNames, ", "));
+        for (NodeManager nodeManager : affectedManagers) {
+            nodeManager.enqueueFault(fault);
+        }
+    }
+
+    public void beginShutdown() {
+        lock.lock();
+        try {
+            this.shutdown = true;
+            cond.signalAll();
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    public void waitForShutdown() {
+        try {
+            this.thread.join();
+        } catch (InterruptedException e) {
+            log.error("Interrupted while waiting for thread shutdown", e);
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    public long startTimeMs() {
+        return startTimeMs;
+    }
+
+    public CoordinatorFaultsResponse getFaults() {
+        Map<String, CoordinatorFaultsResponse.FaultData> faultData = new TreeMap<>();
+        lock.lock();
+        try {
+            getFaultsImpl(faultData, pendingFaults, FaultState.PENDING);
+            getFaultsImpl(faultData, doneFaults, FaultState.DONE);
+        } finally {
+            lock.unlock();
+        }
+        return new CoordinatorFaultsResponse(faultData);
+    }
+
+    private void getFaultsImpl(Map<String, CoordinatorFaultsResponse.FaultData> faultData,
+                               FaultSet faultSet, FaultState state) {
+        for (Iterator<Fault> iter = faultSet.iterateByStart();
+             iter.hasNext(); ) {
+            Fault fault = iter.next();
+            CoordinatorFaultsResponse.FaultData data =
+                new CoordinatorFaultsResponse.FaultData(fault.spec(), state);
+            faultData.put(fault.id(), data);
+        }
+    }
+
+    public void createFault(CreateCoordinatorFaultRequest request) throws ClassNotFoundException {
+        lock.lock();
+        try {
+            Fault fault = FaultSpec.Util.createFault(request.id(), request.spec());
+            pendingFaults.add(fault);
+            cond.signalAll();
+        } finally {
+            lock.unlock();
+        }
+    }
+
+
+    public static void main(String[] args) throws Exception {
+        ArgumentParser parser = ArgumentParsers
+            .newArgumentParser("trogdor-coordinator")
+            .defaultHelp(true)
+            .description("The Trogdor fault injection coordinator");
+        parser.addArgument("--config")
```

Comment:
We have --producer.config and --consumer.config in the other tools. So this should perhaps be --coordinator.config or something like that?